src/typescript-service.ts

```diff
@@ -186,32 +186,24 @@ export class TypeScriptService implements LanguageHandler {
 		});
 	}
 
-	getWorkspaceSymbols(params: rt.WorkspaceSymbolParamsWithLimit): Promise<SymbolInformation[]> {
+	async getWorkspaceSymbols(params: rt.WorkspaceSymbolParamsWithLimit): Promise<SymbolInformation[]> {
 		const query = params.query;
 		const limit = params.limit;
-		return this.projectManager.ensureFilesForWorkspaceSymbol().then(() => {
-			if (!query && this.emptyQueryWorkspaceSymbols) {
-				return this.emptyQueryWorkspaceSymbols;
-			}
 
-			const p = new Promise<SymbolInformation[]>((resolve, reject) => {
-				const configurations = this.projectManager.getConfigurations();
-				const index = 0;
-				const items: SymbolInformation[] = [];
-				this.collectWorkspaceSymbols(query, limit, configurations, index, items, () => {
-					if (!query) {
-						const sortedItems = items.sort((a, b) =>
-							a.name.toLocaleLowerCase().localeCompare(b.name.toLocaleLowerCase()));
-						return resolve(sortedItems);
-					}
-					return resolve(items);
-				});
-			});
-			if (!query) {
-				this.emptyQueryWorkspaceSymbols = p;
-			}
-			return p;
-		});
+		await this.projectManager.ensureFilesForWorkspaceSymbol();
+
+		if (!query && this.emptyQueryWorkspaceSymbols) {
+			return this.emptyQueryWorkspaceSymbols;
+		}
+		const configs = this.projectManager.getConfigurations();
+		const itemsPromise = this.collectWorkspaceSymbols(query, configs);
+		if (!query) {
+			this.emptyQueryWorkspaceSymbols = itemsPromise;
+		}
+		let items = await itemsPromise
+		let sortedItems = items.sort((a, b) => a.name.toLocaleLowerCase().localeCompare(b.name.toLocaleLowerCase()));
+		let limitedItems = sortedItems.slice(0, limit);
```

Comment:
There would be a performance benefit to doing so, but it is minor. The previous limit was 100, which meant as soon as the user entered a query for which there were fewer than 100 results (basically every query other than the empty one), we'd have to do as much work as collecting all items. So the performance benefit would really only be seen for the initial query. I decided to remove this behavior, as it simplifies the code.